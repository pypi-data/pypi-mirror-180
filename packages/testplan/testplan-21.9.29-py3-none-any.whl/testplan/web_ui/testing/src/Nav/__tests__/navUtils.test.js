import React from 'react';
import {StyleSheetTestUtils} from "aphrodite";

import {CreateNavButtons, GetSelectedUid, applyNamedFilter} from '../navUtils';
import {TESTPLAN_REPORT} from '../../Common/sampleReports';
import { PropagateIndices } from '../../Report/reportUtils';

describe('navUtils', () => {

  describe('CreateNavButtons', () => {

    beforeEach(() => {
      // Stop Aphrodite from injecting styles, this crashes the tests.
      StyleSheetTestUtils.suppressStyleInjection();
    });

    afterEach(() => {
      // Resume style injection once test is finished.
      StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    });

    const entries = PropagateIndices(TESTPLAN_REPORT).entries

    it('returns an array of nav buttons', () => {
      const props = {
        breadcrumbLength: 1,
        displayTags: false,
        displayTime: false,
        displayEmpty: true,
        handleNavClick: jest.fn(),
        entries: entries,
        filter: null,
        url: "/testplan/:uid/:selection*"
      }
      const createEntryComponent = jest.fn();

      const navButtons = CreateNavButtons(
        props, createEntryComponent
      );
      expect(navButtons.length).toBe(props.entries.length);
      expect(navButtons).toMatchSnapshot();
    });
  });

  describe('GetSelectedUid', () => {
    it('gets the selected UID', () => {
      const selected = [TESTPLAN_REPORT];
      const uid = GetSelectedUid(selected);
      expect(uid).toBe(TESTPLAN_REPORT.uid);
    });
  });

  describe('FilterFailedOrError', () => {
    it('filters failed or error', () => {
      const entries = [
        {
        "counter": {
                "passed": 1,
                "failed": 0,
                "total": 1,
        },
        "status": "passed",
        },
        {
        "counter": {
            "passed": 0,
            "failed": 1,
            "total": 1,
        },
        "status": "failed",
        },
        {
        "counter": {
            "passed": 0,
            "failed": 0,
            "total": 1,
            "error": 1
        },
        "status": "error",
        }];

      const filteredResult = applyNamedFilter(entries, 'fail');
      const expectedfilteredResult = [
        {
        "counter": {
            "passed": 0,
            "failed": 1,
            "total": 1,
        },
        "status": "failed",
        },
        {
        "counter": {
            "passed": 0,
            "failed": 0,
            "total": 1,
            "error": 1
        },
        "status": "error",
        }];
      expect(filteredResult).toStrictEqual(expectedfilteredResult);
    });
  });

});

