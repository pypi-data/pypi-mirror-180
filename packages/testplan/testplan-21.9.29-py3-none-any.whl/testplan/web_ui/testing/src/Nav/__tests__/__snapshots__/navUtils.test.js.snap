// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navUtils CreateNavButtons returns an array of nav buttons 1`] = `
Array [
  <ListGroupItem
    action={true}
    activeClassName="navButton_1x055n9-o_O-navButtonInteract_kaw7x9-o_O-unselectable_16maxrd-o_O-navButtonInteractFocus_ty7675"
    className="navButton_1x055n9-o_O-navButtonInteract_kaw7x9-o_O-unselectable_16maxrd"
    tabIndex="1"
    tag={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "NavLink",
        "propTypes": Object {
          "activeClassName": [Function],
          "activeStyle": [Function],
          "aria-current": [Function],
          "className": [Function],
          "exact": [Function],
          "innerRef": [Function],
          "isActive": [Function],
          "location": [Function],
          "onClick": [Function],
          "replace": [Function],
          "sensitive": [Function],
          "strict": [Function],
          "style": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/testplan/520a92e4-325e-4077-93e6-55d7091a3f83/21739167-b30f-4c13-a315-ef6ae52fd1f7?"
  />,
  <ListGroupItem
    action={true}
    activeClassName="navButton_1x055n9-o_O-navButtonInteract_kaw7x9-o_O-unselectable_16maxrd-o_O-navButtonInteractFocus_ty7675"
    className="navButton_1x055n9-o_O-navButtonInteract_kaw7x9-o_O-unselectable_16maxrd"
    tabIndex="2"
    tag={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "NavLink",
        "propTypes": Object {
          "activeClassName": [Function],
          "activeStyle": [Function],
          "aria-current": [Function],
          "className": [Function],
          "exact": [Function],
          "innerRef": [Function],
          "isActive": [Function],
          "location": [Function],
          "onClick": [Function],
          "replace": [Function],
          "sensitive": [Function],
          "strict": [Function],
          "style": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/testplan/520a92e4-325e-4077-93e6-55d7091a3f83/8c3c7e6b-48e8-40cd-86db-8c8aed2592c8?"
  />,
]
`;
